### ISSUE TITLE
New packages needed for the next release

### ISSUE THREAD

**Issue by N-thony** [africanmathsinitiative/R-Instat#6562]
- `lemon` for back to back bar chart dialogue #6573

**Comment by shadrackkibet** [comment]
- `mmtable2` for summary tables https://github.com/africanmathsinitiative/R-Instat/pull/6508. Package not on CRAN yet however it is available via a Github repo here https://github.com/ianmoran11/mmtable2

**Comment by rdstern** [comment]
splines2 for smoothing and to replace (I think) splines - if installed, which is withdrawm from CRAN.  I think it is replaced by spline2.  Also the author has written about how this offers periodic splines which we need, see #5911 
Also histdata for interesting and additional data sets, at least one which is used in our main data science book.  
And caret as the main package for machine learning.  Later we should add dialogues, but this is good even if just used with scripts for now.

**Comment by rdstern** [comment]
Also rpivotTable package which will be used on new dialogues in Describe > 2 variables and 3 variables

**Comment by rdstern** [comment]
Yet another package needed is `mlbench` (for machine learning benchmarks!) What an interesting set of datasets.  It is used by the caret package.

**Comment by shadrackkibet** [comment]
- See https://github.com/africanmathsinitiative/R-Instat/issues/6616#issue-946128852

**Comment by rdstern** [comment]
@shadrack - yet another package needed is called survey.  It is the standard r package for the analysis of survey data!  If we want to be respectable for the analysis of survey data, this package is needed.  As with the caret package we should also add (at least) one dialogue.  For the survey package it is a dialogue in the Structured menu It can just be called Survey - Like we have Survival.

**Comment by shadrackkibet** [comment]
@rdstern this is an interesting one to discuss and think through. Let us come up with the design of the dialog and then perhaps good to involve a newbie initially since it is a new dialog. What do you think?

> @shadrack - yet another package needed is called survey. It is the standard r package for the analysis of survey data! If we want to be respectable for the analysis of survey data, this package is needed. As with the caret package we should also add (at least) one dialogue. For the survey package it is a dialogue in the Structured menu It can just be called Survey - Like we have Survival.

**Comment by HawardKetoyoMsatsi** [comment]
@shadrackkibet @rdstern have you had a look at the above comment. Do you have any comments from your side so that the sketch of the dialog could be improved further and interns start working on it.

**Comment by dannyparsons** [comment]
@HawardKetoyoMsatsi @rdstern Thanks for this, but can you move the discussions about the survey package/dialog to a new issue? Then we can keep this discussion for the list of packages that already have functionality and need to be added for the next version.

I will shortly propose a new way of developers adding new package dependences to their PRs, which will make this process clearer, and remove the need to do this through a list in an issue.

**Comment by shadrackkibet** [comment]
`tidytext` https://github.com/africanmathsinitiative/R-Instat/pull/6747
`janitor` https://github.com/africanmathsinitiative/R-Instat/pull/6814

@africanmathsinitiative/developers please remember to always document any new packages in an open issue whenever a new package is added. I am going to pin this issue so that it is easier to find. Thanks

**Comment by dannyparsons** [comment]
I've added the following packages from the discussion here. I've tried to pick out the ones where there is an active dialog that uses them or it was stated as needed for its datasets or had a good case for using from script. Otherwise, e.g. `survey` we should only added when there's a dialog that uses them, as more packages mean larger installation file for users.
- lemon
- HistData
- caret
- rpivotTable
- gcookbook
- tidytext
- janitor
- fastDummies
